oeps/processes/oep-0055/decisions/0001-use-backstage-to-support-maintainers.rst

```diff
@@ -0,0 +1,126 @@
+0001 Use Backstage to Support Maintainers
+#########################################
+
+Status
+******
+
+**Draft**
+
+.. TODO Upon Acceptance
+
+	* Update OEP to mention the catalog-info.ymal file instead of the readme.
+		* Relevant link https://open-edx-proposals.readthedocs.io/en/latest/processes/oep-0055-proc-project-maintainers.html#documenting-maintainers
+
+Context
+*******
+
+* :doc:`OEP-55 <../../oep-0055-proc-project-maintainers>` laid out that we should document the maintainers of a component.
+
+* As a maintainer it is hard to know what components you are maintaining.
+
+* As a community member it is currently difficult to quickly look up who the
+  maintainers are for multiple components quickly.
+
+* Given the number of repos that make up the Open edX platform and the
+  complexity of the job of maintaining them, we need a system for cataloging,
+  and making transparent the critical details of our repositories.
+
+Decision
+********
+
+To solve the above issues and to enable the potential to ease other
+discoverability and maintainership burdens, the Open edX community shall
+maintain a `backstage`_ instance and related metadata.
+
+Consequences
+************
+
+* To support the backstage instance ownership data will have to be stored in a
+  structured way.  We will follow Backstage best practices and store this data
+  in a ``catalog-info.yaml`` file in each component repository.
```

Comment:
While I think Backstage is the right tool for our needs, I feel that we are reimplementing OEP-2, which defines `openedx.yaml` as a way to store repository metadata.

How do you feel about adding a provision this ADR along the lines of: *after X months of trying this decision out, we will either (a) roll it back, or (b) migrate `openedx.yaml` into `catalog-info.yml` and revoke/update OEP-2* ?

Reply:
If the intention is to land this as "Provisional", it inspired me to write this update to OEP-1: https://github.com/openedx/open-edx-proposals/pull/344. Note: the same can be added for ADRs, but I want Livvy's ADR PR to land first: https://github.com/openedx/open-edx-proposals/pull/340.

Reply:
I think right now the `openedx.yaml` file is only really used by the release process.  I think any promise here would mean slating in work to update the release scripts.  It almost feels like a new ADR rather than an addition to this one.  I'm happy to mention the consequence that we'll now have two metadata files with similar data but deciding to get rid of one, seems like another decisions with its own impact.

Reply:
I added a consequence related to this, let me know what you think.

Reply:
> I think right now the openedx.yaml file is only really used by the release process.

In addition to the release metadata, OEP-2 also recommends ["tags" for repos](https://github.com/openedx/open-edx-proposals/blob/master/oeps/best-practices/oep-0002-bp-repo-metadata.rst#tags), which is redundant with the catalog-info spec. Like you say, though, I don't think they're used for anything.

> It almost feels like a new ADR rather than an addition to this one. I'm happy to mention the consequence that we'll now have two metadata files with similar data but deciding to get rid of one, seems like another decisions with its own impact.

I think I am seeing it differently. This feels similar when a legacy frontend is rewritten as an MFE: the decision to write the new MFE and to deprecate the old one ought to go hand-and-hand (unless there's some special reason to the legacy frontend around). Ensuring deprecation of the old thing is the responsibility of the team that adds the new thing.

>  As a result of adding the ``catalog-info.yaml`` file we end with two top-level
  files that have very similar data.  If this decision is not reversed after the
  initial pilot, it may make sense to revisit this consequence and see if we can
  consolidate this metadata into a single file.

This is a good start, but I wish it were more committal. Would you be willing to make a ticket on our backlog to investigate consolidating these two into a single file, and link it here?

Reply:
Yea, I'd be happy to add the ticket, you want it linked in the text of the ADR or just onto this PR?

I see what you're saying about the perspective of it being like the MFEs. In my mind it is not like that because the only real purpose of the openedx.yaml is to indicate what repos are in the official release.  Anything else in there is not actually used in any substantive way.  This new config file has info for maintainership which while related is a different problem to solve so I was seeing them as 2 separate things.

That said, consolidating them should definitely be a long term goal, and I'm happy to put that in writing as a ticket to re-visit this and make a decision in a few months.

Reply:
Great, thanks Feanil.